Configuration load

grishamsc opened the pull request:
- configuration loading from https://api.tonkeeper.com/keys
- caching it
- tests

grishamsc committed:
Configuration load

- add RemoteConfiguration model
- add RemoteConfiguration tests

grishamsc committed:
Configuration load

- add LoadConfigurationRequest
- add LoadConfigurationRequest test

grishamsc committed:
Configuration load

- add ConfigurationLoader
- add ConfigurationLoaderTests

grishamsc committed:
Tests

- remove Tests.swift

grishamsc committed:
Configuration load

- add ConfigurationController
- correct handle error while loading configuration in ConfigurationLoader
- add ConfigurationControllerTests.swift

grishamsc committed:
Configuration load

- add DefaultConfigurationProvider
- add DefaultConfigurationProviderTests

grishamsc committed:
Configuration load

- update ConfigurationController with defaultConfigurationProvider
- update ConfigurationControllerTests

grishamsc committed:
Configuration load

- add DefaultCacheConfigurationProvider
- add DefaultCacheConfigurationProviderTests

grishamsc committed:
Configuration load

- add cacheConfigurationProvider to ConfigurationController
- testConfigurationControllerSaveToCacheAfterLoading test to ConfigurationControllerTests

grishamsc committed:
Configuration load

- attach to loading if get configuration from ConfigurationController while loading in progress
- add test for that case

grishamsc committed:
Configuration load

- update folders structure

oleganza commented on Tests/WalletCoreTests/WalletConfigurationTests/ConfigurationLoaderTests.swift:
```diff
@@ -0,0 +1,119 @@
+    }
+}
+
+extension RemoteConfiguration {
```
I think there is lots of legacy APIs and connections. I would suggest including the minimal URLs we need or even specifying a separate API for "next gen wallet" and sorting things out on the backend to avoid duplication.

grishamsc committed:
Configuration load

- update Package with resources and ton-swift version
- remove all parameters from RemoteConfiguration except tonapiV2Endpoint, tonapiTestnetHost and tonApiV2Key
- update DefaultConfiguration.json
- update tests
